packages/docusaurus-plugin-sitemap/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const fs = require('fs');
+const sitemap = require('sitemap');
+const path = require('path');
+
+const DEFAULT_OPTIONS = {
+  cacheTime: 600 * 1000, // 600 sec - cache purge period
+  changefreq: 'weekly',
+  priority: 0.5,
+};
+
+class DocusaurusPluginSitemap {
+  constructor(opts, context) {
+    this.options = {...DEFAULT_OPTIONS, ...opts};
+    this.context = context;
+  }
+
+  getName() {
+    return 'docusaurus-plugin-sitemap';
+  }
+
+  async createSitemap({siteConfig = {}, routesPaths}) {
+    const {url: hostname} = siteConfig;
+    if (!hostname) {
+      throw new Error(`Url in docusaurus.config.js cannot be empty/undefined`);
+    }
+
+    const urls = routesPaths.map(routesPath => ({
+      url: routesPath,
+      changefreq: this.changefreq,
+      priority: this.priority,
+    }));
+
+    return sitemap
+      .createSitemap({
+        hostname,
+        cacheTime: this.cacheTime,
+        urls,
+      })
+      .toString();
+  }
+
+  async build({siteConfig = {}, routesPaths = [], outDir}) {
```

Comment:
I think a lifecycle of `build` is ambiguous. For sitemap this feels like it should be done after the website is built. Gatsby uses [`onPreBuild`](https://www.gatsbyjs.org/docs/node-apis/#onPreBuild) and [`onPostBuild`](https://www.gatsbyjs.org/docs/node-apis/#onPostBuild) and their sitemap plugin uses [`onPostBuild`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sitemap/src/gatsby-node.js).

How about we call it `siteWillBuild` and `siteDidBuild` (inspired by React component lifecycles)?